src/dune_rules/lib_info.mli

```diff
@@ -86,9 +86,25 @@ module Main_module_name : sig
   type t = Module_name.t option Inherited.t
 end
 
+(** What's the subset of fields that uniquely identifies this stanza? *)
+module Sentinel : sig
+  type t
```

Comment:
What meaning is the name Sentinel is desired to convey here?

Reply:
I called it "sentinel" since this is effectively something that can tell different libraries apart, and uniquely identifies them. in other words, it acts as an indicator of this being the library we're looking for without ambiguity.

Reply:
What's the difference between an id and a sentinel then?

Reply:
no difference. I can change it back if you prefer.

Reply:
I find Id the clearer name, and it seems like other people agree. I'd prefer a name people are more familiar with.

Reply:
absolutely. just changed it back to `library_id`.